bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/LCNBinding.java

```diff
@@ -0,0 +1,2621 @@
+					
+				} else {
+					
+					outputList.add(new LCNOutputElement(theChannel, theBuffer, key));
+					
+				}
+				
+			} else {
+				
+				logger.debug("Channel died!");
+				
+				reconnect(theSocketChannel, theChannel, true);
+				outputList.add(new LCNOutputElement(theChannel, theBuffer, key));
+				
+			}
+			
+		}
+			
+	}
+	
+	/**
+	 * If the binding is activated, open the Selector as well.
+	 */
+	@Override
+	public void activate() {
+
+		try {
+			selector = Selector.open();
+		} catch (IOException e) {
+			logger.error("Unable to open the Selector!");
+		}	
+		
+	}
+
+	/**
+	 * If the binding is deactivated, close the Selector as well.
+	 */
+	@Override
+	public void deactivate() {
+
+		try {
+			selector.close();
+		} catch (IOException e) {
+			logger.error("Unable to close the Selector!");
+		}
+
+	}
+	
+	/**
+	 * Parses input from the LCN bus into usable information for the openHAB system. 
+	 */
+	private void parseInput(ByteBuffer inputBuffer, LCNChannel chan){
+		
+		String input = "";
+		String[] allInput = null;
+		
+		try {
+			
+			allInput = (new String(inputBuffer.array(), SYSTEM_CHARSET).split("\0")[0]).split("\n");
+			input = allInput[0];
+			logger.debug("Channel: " + chan.remote.getHostString() + " received input: " + input);
+			
+		} catch (UnsupportedEncodingException e) {
+			
+			logger.warn("Unable to read input for channel: {}", chan.remote.getHostString());
+			
+		}
+		
+		//Automatically send user name and password if asked for by the system.
+		//Added a priority tag, to ensure that the credentials are send right away.
+		if (input.contains("Username")) {		
+			
+			Credentials cred = null;
+			
+			for (String key : credentialMap.keySet()) {				
+				
+				if (credentialMap.get(key).ip.equals(chan.remote.getHostString()) && credentialMap.get(key).port.equals("" + chan.remote.getPort())) {
+					cred = credentialMap.get(key);
+					break;
+				}
+				
+			}
+			
+			if (null != cred) {
+				quickInternalCommand(LCNSyntax.Command.RAW + "." + cred.username, chan, null, true);
+				logger.debug("Sending username: {}", cred.username);
+			} else {
+				logger.warn("No username found for IP: " + chan.remote.getHostString());
+			}		
+			
+		}
+		
+		if (input.contains("Password")) {	
+			
+			Credentials cred = null;
+			
+			for (String key : credentialMap.keySet()) {
+				
+				if (credentialMap.get(key).ip.equals(chan.remote.getHostString()) && credentialMap.get(key).port.equals("" + chan.remote.getPort())) {
+					cred = credentialMap.get(key);
+					break;
+				}
+				
+			}
+			
+			if (null != cred) {
+				quickInternalCommand(LCNSyntax.Command.RAW + "." + cred.password, chan, null, true);
+				logger.debug("Sending password: {}", cred.password);
+			} else {
+				logger.warn("No password found for IP: " + chan.remote.getHostString());
+			}
+			
+		}
+		
+		//Order information about available segment couplers, right after a connection has been established.
+		if (input.contains("connected")) {
+			
```

Comment:
Wouldn't this also always catch "disconnected" since "connected" is a substring? If so, perhaps re-order the if statements.